crates/core/src/registry.rs

```diff
@@ -224,8 +227,8 @@ pub async fn get_latest_supported_version(dependency: &Dependency) -> Result<Str
             match parse_version_req(dependency.version_req()) {
                 Some(req) => {
                     let new_version = all_versions
-                        .iter()
-                        .find(|version| req.matches(version))
+                        .par_iter()
+                        .find_any(|version| req.matches(version))
```

Comment:
Here we need `find_first` since the versions are sorted. I would benchmark this against `find` as I suspect it might not be beneficial to parallelize here.